### DISCUSSION TITLE
Is there an easy way to tell if dxvk-async is working correctly inside a wine prefix?

### DISCUSSION THREAD

**Discussion by pinbuck** [Sporif/dxvk-async#38] [Q&A]
This is the video guide I used, no errors in terminal but that's not very reassuring:
https://youtu.be/KR7EK2dj3mY

For example, gamemode has a way of testing globally:
![image](https://user-images.githubusercontent.com/96911866/173212563-ad3f8f23-92f7-4b2b-a401-1d9bf54a011f.png)

Mangohud doesnt seem to have anything of the sort.
![image](https://user-images.githubusercontent.com/96911866/173212613-781dfed2-4dec-4999-b4db-825cb192efb5.png)

**Comment by Sporif** [comment]
First you have to rename the state cache file and mesa cache for the game so that they are created from scratch. Then set the environment variable `DXVK_HUD=pipelines` and launch the game. 

Where these files are or where to set the env var depends on how you launch the game i.e (steam, lutris or plain wine from a terminal etc). 

Usually the mesa cache is located at `~/.cache/mesa_shader_cache`. Rename that folder, you don't want to delete it as you'll start from scratch for all games. The state cache file (if nothing else is setting the location) is usually in the game's launch directory. Again, rename the file so you can undo it later.

For steam both caches are stored per game somewhere under `~/.local/share/Steam/steamapps/shadercache/<appid>`.

After launching the game you should see a number in the top left: it's the number of pipelines that have been compiled so far while running the game. If dxvk-async is working it should increase without any or very little stutter whilst playing.

**Comment by pinbuck** [comment]
Thanks for the quick reply, I'll bookmark this for reference.
